### ISSUE TITLE
Tab-completion for multilevel broken on 7.2.0

### ISSUE THREAD

**Issue by xeor** [ipython/ipython#11546]
After `7.2.0`, (enabling of jedi as default), tab completion is broken in multiple places.
See https://github.com/ipython/ipython/issues/11541 and https://github.com/ipython/ipython/issues/11530 for other issues.

I found a new one that bites when using shell_plus in django and django-models..

Example:

**7.1.1**
```
In [1]: import os
In [2]: os.path.<tab>
            os.path.abspath ...

In [2]: User.
              User.DoesNotExist ...

In [2]: User.objects.<tab>
                 User.objects.aggregate ...
```

**7.2.0**
```
In [1]: import os

In [2]: User.<tab>
              User.DoesNotExist ...

In [2]: os.path.<tab>
            os.path.abspath ...

In [2]: User.objects.<tab> (nothing)
```

Putting this in `~/.ipython/profile_default/ipython_config.py` makes `7.2.0` work
```
c = get_config()
c.IPCompleter.use_jedi = False
```

**Comment by cottrell** [comment]
jedi has been a problem in the past. I need to remember and find out again how to disable it completely. Tab-autocomplete broken is like taking my eyes away.

**Comment by prjemian** [comment]
bump, the [suggested workaround](https://github.com/NSLS-II/ophyd/issues/683#issuecomment-469388963) for now is

    ipython --IPCompleter.use_jedi=False

a proper fix would be nicer

**Comment by jorisvandenbossche** [comment]
I think the broken tab completion of accessors in pandas is possibly another example of this: https://github.com/pandas-dev/pandas/issues/26899

**Comment by torbjo** [comment]
Bump / more examples of broken behaviour:

```py
import math
import numpy as np

math.<tab>      # nothing
np.<tab>        # nothing
```

This is really annoying! Thanks for the workaround; thought I had to downgrade.

**Comment by RafalSkolasinski** [comment]
Came across same issue as @torbjo.  tab completion for `np.<tab>` broken in 7.2.0.
Any workaround for it that would work with jupyterlab environment?

**Comment by cottrell** [comment]
Disable Jedi. Just never ever use jedi.

**Comment by RafalSkolasinski** [comment]
Isn't Jedi coming as dependency? 
Shall I just purge it from my environment and Jupyter will fallback to other autocompletion mechanism?

**Comment by RafalSkolasinski** [comment]
Ok, uninstalling it is not a greatest idea as it would remove bunch of other packages:
```
The following packages will be REMOVED:

  adaptive-0.8.1-py37_0
  holoviews-1.12.3-py_2
  ipykernel-5.1.1-py37h24bf2e0_0
  ipython-7.6.1-py37h5ca1d4c_0
  ipywidgets-7.5.0-py_0
  jedi-0.14.0-py37_0
  jupyterlab-0.35.5-py37_0
  jupyterlab_server-0.2.0-py_0
  line_profiler-2.1.2-py37h516909a_1003
  nbdime-1.0.6-py37_0
  notebook-5.7.8-py37_1
  widgetsnbextension-3.5.0-py37_0
```
Can someone share how to set `--IPCompleter.use_jedi=False` for notebook / lab environment from jupyter's docker stack?

**Comment by cottrell** [comment]
I think you can just do the onstart.py or ipython profile config trick as above. You will need a few files from .ipython/profile_default pushed through your docker build.

**Comment by partha1968** [comment]
Can someone share how to set --IPCompleter.use_jedi=False in Windows 10 environment?

**Comment by cottrell** [comment]
@partha1968 I think it should just be in the ipython_config.py whever the profile is on windows

ipython_config.py
3:c.IPCompleter.use_jedi = False

**Comment by davidhalter** [comment]
This is also tracked here: https://github.com/davidhalter/jedi/issues/1299.

It is not working, because it is a safety/security issue. You guys are basically asking for code execution, which can be dangerous. It would be really easy to change this. However I'm not sure it's a good idea (see discussion in the link above).

**Comment by davidhalter** [comment]
This was fixed in davidhalter/jedi#1299.

Though, I would like to point out that this was never a bug, it was a safety feature that should make it less likely that you guys shoot yourself in the foot. Now it's gone and I think that's fine, but it was actually work to "protect" you guys.

Next time, please participate in the discussion in Jedi, I wasn't aware for a long time that this was such a big issue for a lot of people.

Use Jedi 0.15.0.

This can be closed.
